stripe/src/main/java/com/stripe/android/model/SetupIntentParams.java

```diff
@@ -0,0 +1,159 @@
+package com.stripe.android.model;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.stripe.android.utils.ObjectUtils;
+
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class SetupIntentParams implements StripeIntentParams {
+
+    static final String API_PARAM_CLIENT_SECRET = "client_secret";
+    static final String API_PARAM_RETURN_URL = "return_url";
+    static final String API_PARAM_PAYMENT_METHOD_ID = "payment_method";
+
+    @Nullable private String mClientSecret;
+    @Nullable private String mPaymentMethodId;
+    @Nullable private String mReturnUrl;
+
+    private SetupIntentParams() {
+
+    }
+
+    /**
+     * Create the parameters necessary for confirming a PaymentIntent while attaching a
+     * PaymentMethod that already exits.
+     *
+     * @param paymentMethodId the ID of the PaymentMethod that is being attached to the
+     *         PaymentIntent being confirmed
+     * @param clientSecret client secret from the PaymentIntent being confirmed
+     * @param returnUrl the URL the customer should be redirected to after the authorization
+     *         process
+     * @return params that can be use to confirm a PaymentIntent
+     */
+    @NonNull
+    public static SetupIntentParams createConfirmSetupIntenParamsWithPaymentMethodId(
```

Comment:
this should read `createConfirmSetupIntentParamsWithPaymentMethodId` (`Intent` vs `Inten`)